mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoScaler.java

```diff
@@ -398,13 +398,13 @@ public int getDesiredWorkersForScaleDown(final int decrement, final int numCurre
         return desiredWorkers;
       }
 
-      public void scaleDownStage(final int numCurrentWorkers, final int desiredWorkers, final String reason) {
+      public boolean scaleDownStage(final int numCurrentWorkers, final int desiredWorkers, final String reason) {
         logger.info("scaleDownStage decrementing number of workers from {} to {}", numCurrentWorkers, desiredWorkers);
         cancelOutstandingScalingRequest();
         final StageScalingPolicy scalingPolicy = stageSchedulingInfo.getScalingPolicy();
         if (scalingPolicy != null && scalingPolicy.isAllowAutoScaleManager() && !jobAutoscalerManager.isScaleDownEnabled()) {
             logger.warn("Scaledown is disabled for all autoscaling strategy. For stage {} of job {}", stage, jobId);
-            return;
+            return false;
         }
         final Subscription subscription = masterClientApi.scaleJobStage(jobId, stage, desiredWorkers, reason)
```

Comment:
why is this an observable? it feels like there's no need for a stream here?

Reply:
Good point. I didn't check the full logic, so I don't really know.